airflow/utils/timezone.py

```diff
@@ -133,10 +138,10 @@ def make_aware(value: Optional[dt.datetime], timezone: Optional["Timezone"] = No
         value = value.replace(fold=1)
     if hasattr(timezone, 'localize'):
         # This method is available for pytz time zones.
-        return timezone.localize(value)
+        return cast(Any, timezone).localize(value)
```

Comment:
Use #type: ignore instead?

Reply:
There is a discussion about MyPy supporting this case here: https://github.com/python/mypy/issues/1424

One of the nice solutions suggested in the thred as a workaround would translate to:

``
localize = getattr(timezone, 'localize', None)
if localize is not None:
    return localize(value)
``

Reply:
+1 to `getattr`. A less-known fact: For user-defined classes (i.e. not something like `int` or `str`), `hasattr` is actually roughly equivalent to

```python
def hasattr(obj, name):
    try:
        getattr(obj, name)
    except AttributeError:
        return False
    return True
```

So if you need to get the value at some point, `getattr` is always superior to `hasattr`.